src/shared-components/pill-input/PillInput/PillInput.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+import React, {
+    type PropsWithChildren,
+    type JSX,
+    useRef,
+    type KeyboardEventHandler,
+    type HTMLAttributes,
+    type HTMLProps,
+    Children,
+} from "react";
+import classNames from "classnames";
+import { omit } from "lodash";
+import { useMergeRefs } from "react-merge-refs";
+
+import styles from "./PillInput.module.css";
+import { Flex } from "../../utils/Flex";
+
+export interface PillInputProps extends HTMLAttributes<HTMLDivElement> {
+    /**
+     * Callback for when the user presses backspace on an empty input.
+     */
+    onRemoveChildren?: KeyboardEventHandler;
+    /**
+     * Props to pass to the input element.
+     */
+    inputProps?: HTMLProps<HTMLInputElement> & { "data-testid"?: string };
+}
+
+/**
+ * An input component that can contain multiple child elements and an input field.
+ *
+ * @example
+ * ```tsx
+ * <PillInput>
+ *   <div>Child 1</div>
+ *   <div>Child 2</div>
+ * </PillInput>
+ * ```
+ */
+export function PillInput({
+    className,
+    children,
+    onRemoveChildren,
+    inputProps,
+    ...props
+}: PropsWithChildren<PillInputProps>): JSX.Element {
+    const inputRef = useRef<HTMLInputElement>(null);
+    const inputAttributes = omit(inputProps, ["onKeyDown", "ref"]);
+    const ref = useMergeRefs([inputRef, inputProps?.ref]);
+
+    const hasChildren = Children.toArray(children).length > 0;
```

Comment:
Is this different to the following? 
```js
 const hasChildren = !!children
```

Reply:
Yes this is different. If the `children` value is an empty array, `children` is a *positive* value.
In this case, we pass an array of `Pill` as children.